libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1077,17 +1077,21 @@ void AP_InertialSensor::wait_for_sample(void)
         // accel and gyro. This normally completes immediately.
         bool gyro_available = false;
         bool accel_available = false;
-        while (!gyro_available || !accel_available) {
+        while (true) {
```

Comment:
I dont think you really got the issue.

while("explicit check here") will always be an easy code to maintain and check if it can get into a infinite loop.

endless loops depending on breaks in the body can start to get multiple likes checking for different things to decide if it is or it isn't time to break the loop. Relationship between those can get very complex when the code start to receive changes and that happens to occur more in long loop bodies. If take some time to check coverity reports of varios projects you can see that "unreachable code", "invalid checks" and other mistakes start to happen in this kind of code. IOW, silly mistakes start to show up and the consequence of that is -> bugs. When it happens in a endless loops you start to get infinite loops for less testes, less used combinations of the checks in those bodies. And this always start with a commit adding a endless loop statement.